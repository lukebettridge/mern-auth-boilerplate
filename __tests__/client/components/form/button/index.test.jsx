import React from "react";
import renderer from "react-test-renderer";
import { shallow } from "enzyme";

import Button from "components/form/button";

describe("Button component", () => {
	it("snapshot renders", () => {
		const component = renderer.create(<Button />);
		const tree = component.toJSON();

		expect(tree).toMatchSnapshot();
	});

	it("renders children", () => {
		const wrapper = shallow(
			<Button>
				<p>Text</p>
			</Button>
		);

		expect(wrapper.find("p").length).toEqual(1);
		expect(wrapper.find("p").text()).toEqual("Text");
	});
});
