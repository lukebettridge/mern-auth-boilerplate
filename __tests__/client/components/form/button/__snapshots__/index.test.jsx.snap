// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component snapshot renders 1`] = `
<styles__Button
  type="button"
/>
`;
